data/src/main/java/dk/nodes/template/data/network/RestPostRepository.kt

```diff
@@ -7,38 +7,31 @@ import kotlinx.coroutines.channels.BroadcastChannel
 import kotlinx.coroutines.channels.Channel
 import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.asFlow
+import retrofit2.Response
 import javax.inject.Inject
 
 class RestPostRepository @Inject constructor(private val api: Api) : PostRepository {
 
-    private val postsChannel = BroadcastChannel<List<Post>>(Channel.CONFLATED)
-        .also {
-            it.offer(listOf())
-        }
+    private val postsChannel = BroadcastChannel<List<Post>>(Channel.CONFLATED).apply {
+        offer(listOf())
+    }
 
     override fun getPostsFlow(): Flow<List<Post>> {
         return postsChannel.asFlow()
     }
 
-    override suspend fun getPosts(): List<Post> {
-        val response = api.getPosts()
-        if (response.isSuccessful) {
-
-            return response.body()
-                ?.also {
-                    postsChannel.send(it)
-                }
-                ?: throw(RepositoryException(
-                    response.code(),
-                    response.errorBody()?.string(),
-                    response.message()
-                ))
-        } else {
-            throw(RepositoryException(
-                response.code(),
-                response.errorBody()?.string(),
-                response.message()
-            ))
+    override suspend fun getPosts(): List<Post> = with(api.getPosts()) {
+        return result.also {
+            postsChannel.send(it)
         }
     }
+
+    private val <T> Response<T>.result: T
```

Comment:
I'm not sure about this, throwing an exception on a field should not be the expected behavior

Reply:
agree 🙈, maybe just have it as a function?

```kotlin
    @Throws(RepositoryException::class)
    private fun <T> Response<T>.result(): T = body()?.let {
        return it
    } ?: throw(RepositoryException(
            code = code(),
            errorBody = errorBody()?.string(),
            msg = message()
    ))
```

Reply:
prolly should also be available to other remote repository impls

Reply:
I'd call it `getResultOrThrow()` then